### ISSUE TITLE
Only show link/file popover menu on mouseup

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/text#1059]
Currently the link/file menu is shown during the selection process already, which can be annoying if you select text from the bottom to the top.

We could make sure the menu only opens on mouseup – which is also what Medium does.

**Comment by leobard** [comment]
I agree. I noticed multiple problems with linking in v3.1.0
- I saw no "link" button in the format menu. There is an "insert image" button though. I would expect the same for "insert link"
- Selecting text suddenly opens a confusing "add link | link file" menu. This is confusing me because:
  - When selecting text, I expect a popup menu will ALL actions I can do on the text.
  - I only can create a link. What about bold/italic/...?
- once you have created the link, it is immediately opened when clicking it with left-mouse-button. This is confusing because:
  - I assume I am in a text editor. When I click text, I may want to place a cursor. Opening the link is confusing.
  - I wanted to left-click the link to select it with the next step being right-click to edit

I refer to other text editors which I use and which are very common:
- Microsoft Word - right-click shows what you can do with text
- Typora - right-click shows what you can do with text
- Confluence - you click on a link and it shows a popup menu
- This editor here on github - you enter a link and it has no menu at all. You can edit the link, click it, select it, trouble-free. It is plaintext though, but whatever, I am fine

Proposed solution
- add a "add link" button to the format menu
- when clicking a link, show a small popup (comparable to the one now visible when selecting text) with the options "open link", "edit link", "remove link".
- add a shortcut strg+leftmouseclick to open a link

**Comment by leobard** [comment]
@jancborchardt if you agree to my proposal above, I have an unusal offer:

I could play around with the editor and get the behaviour implemented I described above and submit a pull request/diff.

I have never pushed anything into NC though, so I would need help to setup the development environment for the text.app and I have to learn the way the text.app maintainers want diffs from noobs. If someone could spend 30-60 minutes with me during an evening in the coming week, helping me understand how dev works, and if the code is organized in a way I understand, I could look into this and try to get into into the form I suggested.
